src/main/java/randoop/generation/CoverageTracker.java

```diff
@@ -0,0 +1,217 @@
+   * under test.
+   *
+   * @param classesUnderTest names of all the classes under test
+   */
+  public CoverageTracker(Set<String> classesUnderTest) {
+    this.classesUnderTest = new HashSet<>(classesUnderTest);
+    this.instrumentingClassLoader = new InstrumentingClassLoader(this);
+    this.instrumenter = new Instrumenter(runtime);
+    this.data = new RuntimeData();
+
+    try {
+      this.runtime.startup(data);
+    } catch (Exception e) {
+      e.printStackTrace();
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Attempts to instrument a class and return the byte array representation.
+   *
+   * @param className name of the class to instrument
+   * @return the instrumented bytes of the class, or null if the class does not need to be
+   *     instrumented
+   */
+  public byte[] instrumentClass(String className) {
+    // Only instrument classes under test and their nested classes.
+    if (!shouldBeInstrumented(className)) {
+      return null;
+    }
+
+    final byte[] instrumented;
+    final String resource = getResourceFromClassName(className);
+    InputStream original = getClass().getResourceAsStream(resource);
+
+    if (original == null) {
+      System.err.println("No resource with name: " + resource + " found by CoverageTracker!");
+      System.exit(1);
+    }
+
+    try {
+      // Instrument the class so that branch coverage data can be obtained.
+      instrumented = instrumenter.instrument(original, className);
+      original.close();
+    } catch (IOException e) {
+      e.printStackTrace();
+      System.exit(1);
+      throw new Error("This can't happen.");
+    }
+
+    return instrumented;
+  }
+
+  /**
+   * Return true if the given class should be instrumented.
+   *
+   * @param className the name of the class to check
+   * @return true iff the given class is a clall under test or a nested class of a class under test
+   */
+  private boolean shouldBeInstrumented(String className) {
+    if (classesUnderTest.contains(className)) {
+      return true;
+    }
+    int dollarPos = className.indexOf('$');
+    if (dollarPos == -1) {
+      return false;
+    }
+
+    String outerClassName = className.substring(0, dollarPos);
+    return classesUnderTest.contains(outerClassName);
+  }
+
+  /**
+   * Instruments and then loads the class with the given name.
+   *
+   * @param className name of the class
+   * @return {@code Class} object that has been instrumented for coverage data collection. Returns
+   *     null if class with target name cannot be found.
+   */
+  public Class<?> instrumentAndLoadClass(String className) {
+    try {
+      return instrumentingClassLoader.loadClass(className);
+    } catch (ClassNotFoundException e) {
+      e.printStackTrace();
+      System.exit(1);
+      throw new Error("This can't happen.");
+    }
+  }
+
+  /**
+   * Summarizes coverage information for all methods under test. At this point, Jacoco has already
```

Comment:
Jacoco inserts "probes" into the instrumented classes to gather coverage information. This coverage information is updated immediately as the methods in these classes are executed. However, the CoverageTracker's view of the coverage information isn't updated unless we explicitly ask Jacoco for the information. Thus, I named it "summarize" because new information isn't being computed/created, rather we're summarizing the information given to us by Jacoco.  I do agree that updateBranchCoverageMap is more clear though.